server/app/models.py

```diff
@@ -180,6 +180,34 @@ def get_final_submission(self, assignment_key):
                 FinalSubmission.assignment==assignment_key,
                 FinalSubmission.submitter==self.key).get()
 
+    def get_submission_before(self, assignment_key, before_time):
+        """Get this users last submission before some server time"""
+        if isinstance(assignment_key, Assignment):
+            assignment_key = assignment_key.key
+        group = self.get_group(assignment_key)
+        if group and self.key in group.member:
+            subms = Submission.query(
+                Submission.submitter.IN(group.member),
+                Submission.assignment == assignment_key,
+                Submission.server_time < before_time).order(
+                  -Submission.server_time
+                ).fetch(100)
+        else:
+            subms = Submission.query(
+                Submission.submitter ==self.key,
+                Submission.assignment == assignment_key,
+                Submission.server_time < before_time,
+                ).order(
+                  -Submission.server_time
```

Comment:
This `-` is worth investigation - it  might actually do the opposite of what we want in production. This works in testing because we don't have > 100 submissions.